### ISSUE TITLE
[Bug] Relationship field broken - underscore in field name

### ISSUE THREAD

**Issue by dividy** [Laravel-Backpack/CRUD#2874]
# Bug report
Relationship field seems broken if you have an underscore in the name of the field.

### What I did
```php
$this->crud->addFields([
    [
        'label' => 'Billing address',
        'type' => "relationship",
        'name' => 'billing_address',
        'attribute' => 'full_address',
        'ajax' => true,
        'inline_create' => [ 
            'entity' => 'address',
            'force_select' => true,
            'modal_class' => 'modal-dialog modal-xl',
        ]
    ],
]);
```

### What I expected to happen
A good route generated : /site/fetch/billing_address

### What happened
A bad route has been generated : /site/fetch/billing-address

### Backpack, Laravel, PHP, DB version
When I run ```php artisan backpack:version``` the output is:

### PHP VERSION:
PHP 7.2.19 (cli) (built: May 29 2019 13:58:59) ( ZTS MSVC15 (Visual C++ 2017) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

### LARAVEL VERSION:
v7.10.3@6e927e78aafd578d59c99608e7f0e23a5f7bfc5a

### BACKPACK VERSION:
4.1.3@02988a0d14b09e55949781d729060df910f0d94b

**Comment by dividy** [comment]
In method **setupFetchOperationRoutes** of **FetchOperation**, I can see : 

```php
protected function setupFetchOperationRoutes($segment, $routeName, $controller)
    {
        // get all method names on the current model that start with "fetch" (ex: fetchCategory)
        // if a method that looks like that is present, it means we need to add the routes that fetch that entity
        preg_match_all('/(?<=^|;)fetch([^;]+?)(;|$)/', implode(';', get_class_methods($this)), $matches);

        if (count($matches[1])) {
            foreach ($matches[1] as $methodName) {
                Route::post($segment.'/fetch/'.Str::kebab($methodName), [
                    'uses'      => $controller.'@fetch'.$methodName,
                    'operation' => 'FetchOperation',
                ]);
            }
        }
    }
```

**Comment by dividy** [comment]
I thought problem could be in the field template, so I had a look at **relationship/fetch.blade.php** : 
```php    
$routeEntity = Str::kebab($entityWithoutAttribute);
```
and

```php
$field['data_source'] = $field['data_source'] ?? url($crud->route.'/fetch/'.$routeEntity);
```

**Comment by pxpm** [comment]
Hello @dividy, thanks for report.

What's your fetch function definition ? 

Best,

**Comment by dividy** [comment]
```php
protected function fetchBillingAddress()
    {
        return $this->fetch([
            'model' => \App\Models\Address::class,
            'searchable_attributes' => ['name'],
            'paginate' => 10,
        ]);
    }
```

**Comment by dividy** [comment]
If i had this parameters to the relationship field, it works: 

> 'data_source' => url($this->crud->route.'/fetch/billing-address'),

**Comment by dividy** [comment]
I also, there is something strange when updating.

It tried to update the field "address", instead of "address_id" and the request validation doesn't work either...

**Comment by dividy** [comment]
Ok, managed to get it working this way  : (still don't know why it works this way)
```php
[
                'label' => 'Adresse',
                'type' => "relationship",
                'name' => 'address_id',
                'entity' => 'address',
                'attribute' => 'full_address',
                'ajax' => true,
                'placeholder' => "Choisissez une adresse",
                'inline_create' => true
            ],
            [
                'label' => 'Adresse de facturation',
                'type' => "relationship",
                'name' => 'billing_address_id',
                'entity' => 'address',
                'data_source' => url($this->crud->route.'/fetch/billing-address'),
                'attribute' => 'full_address',
                'ajax' => true,
                'placeholder' => "Choisissez une adresse",
                'inline_create' => true
            ],
```

**Comment by pxpm** [comment]
> I also, there is something strange when updating.
> 
> It tried to update the field "address", instead of "address_id" and the request validation doesn't work either...

This is expected behaviour, and I think it's in docs. If you use relationship, you validate also with relationship name, in your scenario would validate `address => 'required'`

I will have a look and investigate if we really need to Str::kebab here. 

Once again, thank you @dividy ! \o

**Comment by dividy** [comment]
Yes, thanks.

However, I still have a bug...

When I create an item.

I don't select an Address, however I select a Billing Address.
Backpack saves the Billing Address in both fields instead of saving NULL in Address field. (Update works fine tho).

I thought I had to change the 'entity' property, which is 'address' for both to 'address' and 'billing_address' respectively, but then I get this error : Route [billing_address-inline-create-save] not defined. 


Do you have an idea ? ;-)

**Comment by dividy** [comment]
Any clue @pxpm ?

**Comment by dividy** [comment]
Ok @pxpm ,

Managed to fix it by replacing **Str::kebab** with **Str::snake** 
(Str::kebab replaces underscores by hyphens)

**Comment by tabacitu** [comment]
Fixed by #2905 so let's move the conversation there please. Thanks a lot for raising the issue @dividy ! Very good collaboration between you and @pxpm here I see.
